pkgs/development/python-modules/scancode-toolkit/default.nix

```
@@ -0,0 +1,112 @@
+{ lib
+, fetchPypi
+, buildPythonPackage
+, isPy3k
+, markupsafe
+, click
+, typecode
+, gemfileparser
+, pefile
+, fingerprints
+, spdx-tools
+, fasteners
+, pycryptodome
+, urlpy
+, dparse
+, jaraco_functools
+, pkginfo
+, debian-inspector
+, extractcode
+, ftfy
+, pyahocorasick
+, colorama
+, jsonstreams
+, packageurl-python
+, pymaven-patch
+, nltk
+, pygments
+, bitarray
+, jinja2
+, javaproperties
+, boolean-py
+, license-expression
+, extractcode-7z
+, extractcode-libarchive
+, typecode-libmagic
+}:
+let
+  bitarray' = bitarray.overridePythonAttrs ({ ... }: rec {
```

```suggestion
  bitarray_0_9 = bitarray.overridePythonAttrs ({ ... }: rec {
```

Also I think in pythonPackages moving it to top-level is preferred.

---

Overrides are not allowed within `pythonPackages`. Is this ever used in a way other than the command line tool? If not, you should move it to `pkgs/tools/...` and use `buildPythonApplication`.

---

@dotlambda So, are you saying this is fine as-is? Or that I'd need to make a new `bitarray_0_9` package (without it being an override)? (I'm unsure if this will ever be used as a library, but it has a way to be imported so I'm going to assume "yes" in some scenarios.)

---

It's not fine and neither would be `pythonPackages.bitarray_0_9`. "Fixing" angr in https://github.com/NixOS/nixpkgs/pull/120113.

---

I don't know why I thought this was necessary to begin with, but I was able to build by just removing the version bound in the `setup.cfg`. We'll see if this has any repercussions.